messaging/mqtt/configuration.go

```diff
@@ -0,0 +1,131 @@
+/********************************************************************************
+ *  Copyright 2020 Dell Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *******************************************************************************/
+
+// Package mqtt provides additional functionality to aid in configuring a MQTT client.
+package mqtt
+
+import "strconv"
+
+// mqttOptionalConfigurationBuilder encapsulates the optional configuration data following the builder pattern. Updating
+// values are done via the exported builder methods and the final map can be obtained by calling the Build method.
+//
+// This is the recommended way to create optional parameters for an MQTT client since the underlying structure is a map
+// of string : string which can introduce issues with type casting. The builder methods for this struct are created to
+// accept the expected types to reduce the chance of incorrect type casting.
+type mqttOptionalConfigurationBuilder struct {
+	options map[string]string
+}
+
+// NewMQTTOptionalConfigurationBuilder constructs a new mqttOptionalConfigurationBuilder with the necessary properties
+// set.
+func NewMQTTOptionalConfigurationBuilder(topic string) *mqttOptionalConfigurationBuilder {
```

Comment:
Why topic specified? It has been captured in TopicChannel for subscribing and service specific config for publishing.. I.e. doesn't belong in Optional

Reply:
Good catch. Turns out I was using this for another option `SetWill` which required the topic information at the time of client creation. I removed that configuration option as well as this one since it is not being used.